### ISSUE TITLE
Problem: "Upload DIP" task restarts when mcp-server is restarted

### ISSUE THREAD

**Issue by scollazo** [archivematica/Issues#112]
**Expected behaviour**
After a restart of mcp-server, old transfers shouldn't be active again

**Current behaviour**
After restarting MCP server, the "Upload dip" question gets asked again for finished transfers

**Steps to reproduce**
1. Run a transfer
2. Restart archivematica-mcp-server

**Your environment (version of Archivematica, OS version, etc)**
Archivematica 1.7 installed using ansible on ubuntu trusty

**Comment by ross-spencer** [comment]
A number of tasks are triggered when the MCP server is restarted. This shouldn't happen. If we truncate the `Tasks` table and then restart, we can see which ones:

taskUUID | createdTime | fileUUID | fileName | exec | arguments | startTime | endTime | client | stdOut | stdError | exitCode | jobuuid
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
168a4e49-5aea-4689-bde3-72af0f0a8e85 | 2018-08-22 21:23:46.605730 |   | 1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb | getAipStorageLocations_v0.0 | DS | 2018-08-22 21:23:46.617624 | 2018-08-22 21:23:46.978150 | 87e32a1a80c4_1 | {'Default location': '/api/v2/location/default/DS/', u'Store DIP in standard Archivematica Directory': u'/api/v2/location/52efb953-1052-432a-a552-f2ff377cfbf6/'} | WARNING 2018-08-22 21:23:46 py.warnings:constant_time:<module>:26: /usr/local/lib/python2.7/dist-packages/cryptography/hazmat/primitives/constant_time.py:26: CryptographyDeprecationWarning: Support for your Python version is deprecated. The next version of cryptography will remove support. Please upgrade to a 2.7.x release that supports hmac.compare_digest as soon as possible. utils.DeprecatedIn23, getAipStorageLocations.py: DEBUG 2018-08-22 21:23:46,925 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service getAipStorageLocations.py: DEBUG 2018-08-22 21:23:46,932 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "GET /api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/ HTTP/1.1" 200 None getAipStorageLocations.py: DEBUG 2018-08-22 21:23:46,935 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service getAipStorageLocations.py: DEBUG 2018-08-22 21:23:46,946 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "GET /api/v2/location/?pipeline__uuid=9325dcbe-5a2e-4494-95fa-328bdd8da61c&purpose=DS&offset=0 HTTP/1.1" 200 None getAipStorageLocations.py: DEBUG 2018-08-22 21:23:46,946 root:get_aip_storage_locations:19: Storage Directories: [{u'uuid': u'52efb953-1052-432a-a552-f2ff377cfbf6', u'pipeline': [u'/api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/'], u'used': u'252807', u'description': u'Store DIP in standard Archivematica Directory', u'space': u'/api/v2/space/20ade3d8-e278-4f1d-97fc-083f82ad6c8e/', u'enabled': True, u'quota': None, u'relative_path': u'var/archivematica/sharedDirectory/www/DIPsStore', u'purpose': u'DS', u'path': u'/var/archivematica/sharedDirectory/www/DIPsStore', u'resource_uri': u'/api/v2/location/52efb953-1052-432a-a552-f2ff377cfbf6/'}] | 0 | 28959ab8-bce8-4280-b628-593708c0eaf5
39fb3759-51c2-4eb5-9c92-63ab66ceb37f | 2018-08-22 21:23:46.185765 |   | 1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb | upload-qubit_v0.0 | --url="http://localhost/atom" --email="demo@example.com" --password="demo" --uuid="257ff562-2cdc-48ca-8f26-c6fd37f9f1fb" --debug="" --version="2" --rsync-command="" --rsync-target="" | 2018-08-22 21:23:46.197887 | 2018-08-22 21:23:46.507355 | 87e32a1a80c4_8 |   | /usr/local/lib/python2.7/dist-packages/cryptography/hazmat/primitives/constant_time.py:26: CryptographyDeprecationWarning: Support for your Python version is deprecated. The next version of cryptography will remove support. Please upgrade to a 2.7.x release that supports hmac.compare_digest as soon as possible. utils.DeprecatedIn23, [uploadDIP] No target was selected | 1 | 4cfe7dc6-dc25-4df6-8e86-6bec6f16ad14
491260bd-9398-4667-951a-6b1564beceaa | 2018-08-22 21:23:46.527939 |   | 1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb | copy_v0.0 | "%sharedPath%watchedDirectories/uploadDIP/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb/" "/var/archivematica/sharedDirectory/watchedDirectories/uploadedDIPs/." -R | 2018-08-22 21:23:46.536876 | 2018-08-22 21:23:46.550342 | 87e32a1a80c4_6 |   |   | 0 | 3d7ce1e7-4f36-45c4-a14a-4bd4844421c1
6f478aac-6dac-4b74-a9bb-dcae923673d7 | 2018-08-22 21:23:46.268172 |   | 1-1171e3e5-e03f-4aa1-8a5f-6d910ccf4d93 | checkTransferDirectoryForObjects_v0.0 | "/var/archivematica/sharedDirectory/watchedDirectories/SIPCreation/completedTransfers/1-1171e3e5-e03f-4aa1-8a5f-6d910ccf4d93/objects/" | 2018-08-22 21:23:46.285051 | 2018-08-22 21:23:46.312076 | 87e32a1a80c4_1 |   |   | 0 | 7508acdf-b7db-43fd-a764-48550d668f84
f83b1549-23c3-4c56-be28-84d84e247503 | 2018-08-22 21:23:47.062058 |   | 1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb | storeAIP_v0.0 | "/api/v2/location/default/DS/" "/var/archivematica/sharedDirectory/watchedDirectories/uploadedDIPs/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb" "257ff562-2cdc-48ca-8f26-c6fd37f9f1fb" "1" "DIP" | 2018-08-22 21:23:47.072910 | 2018-08-22 21:23:47.761621 | 87e32a1a80c4_5 | Checking if DIP 37fc55d3-ddf2-488f-ad35-e868d3e661c9 parent AIP has been created... Parent AIP exists so relationship can be created. Storage service created DIP: {u'size': 84269, u'status': u'UPLOADED', u'origin_path': u'watchedDirectories/uploadedDIPs/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb/', u'origin_pipeline': u'/api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/', u'uuid': u'37fc55d3-ddf2-488f-ad35-e868d3e661c9', u'related_package_uuid': u'257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', u'replicas': [], u'encrypted': False, u'current_path': u'37fc/55d3/ddf2/488f/ad35/e868/d3e6/61c9/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', u'replicated_package': None, u'origin_location': u'/api/v2/location/8829fd7d-2e10-4b7c-acbb-9a8d799daabd/', u'aip_subtype': u'Archival Information Package', u'misc_attributes': {}, u'current_full_path': u'/var/archivematica/sharedDirectory/www/DIPsStore/37fc/55d3/ddf2/488f/ad35/e868/d3e6/61c9/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', u'current_location': u'/api/v2/location/52efb953-1052-432a-a552-f2ff377cfbf6/', u'agents': [], u'related_packages': [u'/api/v2/file/257ff562-2cdc-48ca-8f26-c6fd37f9f1fb/'], u'resource_uri': u'/api/v2/file/37fc55d3-ddf2-488f-ad35-e868d3e661c9/', u'events': [], u'package_type': u'DIP'} | WARNING 2018-08-22 21:23:47 py.warnings:constant_time:<module>:26: /usr/local/lib/python2.7/dist-packages/cryptography/hazmat/primitives/constant_time.py:26: CryptographyDeprecationWarning: Support for your Python version is deprecated. The next version of cryptography will remove support. Please upgrade to a 2.7.x release that supports hmac.compare_digest as soon as possible. utils.DeprecatedIn23, storeAIP.py: DEBUG 2018-08-22 21:23:47,407 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service storeAIP.py: DEBUG 2018-08-22 21:23:47,415 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "GET /api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/ HTTP/1.1" 200 None storeAIP.py: DEBUG 2018-08-22 21:23:47,418 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service storeAIP.py: DEBUG 2018-08-22 21:23:47,430 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "GET /api/v2/location/?pipeline__uuid=9325dcbe-5a2e-4494-95fa-328bdd8da61c&purpose=CP&offset=0 HTTP/1.1" 200 None storeAIP.py: DEBUG 2018-08-22 21:23:47,433 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service storeAIP.py: DEBUG 2018-08-22 21:23:47,447 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "GET /api/v2/file/?uuid=257ff562-2cdc-48ca-8f26-c6fd37f9f1fb&offset=0 HTTP/1.1" 200 None storeAIP.py: DEBUG 2018-08-22 21:23:47,455 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service storeAIP.py: DEBUG 2018-08-22 21:23:47,463 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "GET /api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/ HTTP/1.1" 200 None storeAIP.py: INFO 2018-08-22 21:23:47,463 archivematica.common:create_file:283: Creating file with {'current_path': '1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', 'related_package_uuid': '257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', 'agents': [], 'current_location': '/api/v2/location/default/DS/', 'size': 84269, 'origin_path': u'watchedDirectories/uploadedDIPs/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb/', 'origin_pipeline': u'/api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/', 'uuid': '37fc55d3-ddf2-488f-ad35-e868d3e661c9', 'package_type': 'DIP', 'origin_location': u'/api/v2/location/8829fd7d-2e10-4b7c-acbb-9a8d799daabd/', 'aip_subtype': 'Archival Information Package', 'events': []} storeAIP.py: DEBUG 2018-08-22 21:23:47,466 urllib3.connectionpool:_new_conn:208: Starting new HTTP connection (1): archivematica-storage-service storeAIP.py: DEBUG 2018-08-22 21:23:47,723 urllib3.connectionpool:_make_request:396: http://archivematica-storage-service:8000 "POST /api/v2/file/ HTTP/1.1" 201 None storeAIP.py: INFO 2018-08-22 21:23:47,724 archivematica.common:create_file:299: Status code of create file/package request: 201 storeAIP.py: INFO 2018-08-22 21:23:47,725 archivematica.mcp.client.storeAIP:store_aip:198: Storage service created DIP: {u'size': 84269, u'status': u'UPLOADED', u'origin_path': u'watchedDirectories/uploadedDIPs/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb/', u'origin_pipeline': u'/api/v2/pipeline/9325dcbe-5a2e-4494-95fa-328bdd8da61c/', u'uuid': u'37fc55d3-ddf2-488f-ad35-e868d3e661c9', u'related_package_uuid': u'257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', u'replicas': [], u'encrypted': False, u'current_path': u'37fc/55d3/ddf2/488f/ad35/e868/d3e6/61c9/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', u'replicated_package': None, u'origin_location': u'/api/v2/location/8829fd7d-2e10-4b7c-acbb-9a8d799daabd/', u'aip_subtype': u'Archival Information Package', u'misc_attributes': {}, u'current_full_path': u'/var/archivematica/sharedDirectory/www/DIPsStore/37fc/55d3/ddf2/488f/ad35/e868/d3e6/61c9/1-257ff562-2cdc-48ca-8f26-c6fd37f9f1fb', u'current_location': u'/api/v2/location/52efb953-1052-432a-a552-f2ff377cfbf6/', u'agents': [], u'related_packages': [u'/api/v2/file/257ff562-2cdc-48ca-8f26-c6fd37f9f1fb/'], u'resource_uri': u'/api/v2/file/37fc55d3-ddf2-488f-ad35-e868d3e661c9/', u'events': [], u'package_type': u'DIP'} | 0 | 4272daf8-fd69-459c-9ea1-2db1d4fc9e09

**Comment by ross-spencer** [comment]
**NB.** trying to verify how far this issue dates back, I run into problems deploying with Docker:
```
Step 5/22 : ENV PYTHONUNBUFFERED 1
 ---> Using cache
 ---> f789f3bcdfb3
Step 6/22 : RUN set -ex 	&& apt-get update 	&& apt-get install -y --no-install-recommends 		gettext 		libmysqlclient-dev 		libldap2-dev 	libsasl2-dev 		locales 		locales-all 	&& rm -rf /var/lib/apt/lists/*
 ---> Running in 64d0404b6b3a
+ apt-get update
Get:1 http://security.debian.org/debian-security stretch/updates InRelease [94.3 kB]
Ign:2 http://deb.debian.org/debian stretch InRelease
Get:3 http://deb.debian.org/debian stretch-updates InRelease [91.0 kB]
Get:4 http://deb.debian.org/debian stretch Release [118 kB]
Get:5 http://deb.debian.org/debian stretch Release.gpg [2434 B]
Get:6 http://security.debian.org/debian-security stretch/updates/main amd64 Packages [391 kB]
Get:7 http://deb.debian.org/debian stretch-updates/main amd64 Packages [5148 B]
Get:8 http://deb.debian.org/debian stretch/main amd64 Packages [7099 kB]
Fetched 7801 kB in 4s (1732 kB/s)
Reading package lists...
+ apt-get install -y --no-install-recommends gettext libmysqlclient-dev libldap2-dev libsasl2-dev locales locales-all
Reading package lists...
Building dependency tree...
Reading state information...
Package libmysqlclient-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libmysqlclient-dev' has no installation candidate
ERROR: Service 'archivematica-mcp-server' failed to build: The command '/bin/sh -c set -ex 	&& apt-get update 	&& apt-get install -y --no-install-recommends 		gettext libmysqlclient-dev 		libldap2-dev 		libsasl2-dev 		locales 		locales-all 	&& rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100
```
Will focus on looking at `1.7.2`

**Comment by ross-spencer** [comment]
**Deleting the contents of the Watched Directories**

After deletion:
```
.
├── arrange
├── completed
│   └── transfers
├── currentlyProcessing
├── DIPbackups
├── failed
├── rejected
├── sharedMicroServiceTasksConfigs
│   ├── createXmlEventsAssist
│   ├── generateAIP
│   │   └── bagit
│   ├── processingMCPConfigs
│   │   ├── automatedProcessingMCP.xml
│   │   └── defaultProcessingMCP.xml
│   └── transcoder
│       └── defaultIcons
├── SIPbackups
├── tmp
├── watchedDirectories
│   ├── activeTransfers
│   │   ├── baggitDirectory
│   │   ├── baggitZippedDirectory
│   │   ├── Dspace
│   │   ├── maildir
│   │   ├── standardTransfer
│   │   └── TRIM
│   ├── approveNormalization
│   ├── approveSubmissionDocumentationIngest
│   ├── quarantined
│   ├── SIPCreation
│   │   ├── completedTransfers
│   │   └── SIPsUnderConstruction
│   ├── storeAIP
│   ├── system
│   │   ├── autoProcessSIP
│   │   ├── autoRestructureForCompliance
│   │   ├── createAIC
│   │   └── reingestAIP
│   ├── uploadDIP
│   ├── uploadedDIPs
│   └── workFlowDecisions
│       ├── compressionAIPDecisions
│       ├── createDip
│       ├── createTree
│       ├── examineContentsChoice
│       ├── extractPackagesChoice
│       ├── metadataReminder
│       ├── quarantineTransfer
│       ├── selectFormatIDToolIngest
│       └── selectFormatIDToolTransfer
└── www
    ├── AIPsStore
    │   ├── e022
    │   │   └── 606a
    │   │       └── b9ce
    │   │           └── 480a
    │   │               └── ba3a
    │   │                   └── 0199
    │   │                       └── 13f4
    │   │                           └── 998a
    │   │                               └── 2-e022606a-b9ce-480a-ba3a-019913f4998a.7z
    │   └── transferBacklog
    │       ├── arrange
    │       └── originals
    ├── DIPsStore
    │   ├── 466c
    │   │   └── 7dc8
    │   │       └── e92d
    │   │           └── 46f2
    │   │               └── a7b7
    │   │                   └── b9af
    │   │                       └── 2e08
    │   │                           └── d990
    │   │                               └── 2-e022606a-b9ce-480a-ba3a-019913f4998a
    │   │                                   ├── METS.e022606a-b9ce-480a-ba3a-019913f4998a.xml
    │   │                                   ├── objects
    │   │                                   │   └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d-one_file.txt
    │   │                                   ├── processingMCP.xml
    │   │                                   └── thumbnails
    │   │                                       └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d.jpg
    │   └── 9b18
    │       └── b071
    │           └── b67b
    │               └── 40c9
    │                   └── 895d
    │                       └── 2f59
    │                           └── 5c26
    │                               └── 7a37
    │                                   └── 2-e022606a-b9ce-480a-ba3a-019913f4998a
    │                                       ├── METS.e022606a-b9ce-480a-ba3a-019913f4998a.xml
    │                                       ├── objects
    │                                       │   └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d-one_file.txt
    │                                       ├── processingMCP.xml
    │                                       └── thumbnails
    │                                           └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d.jpg
    └── thumbnails
        └── e022606a-b9ce-480a-ba3a-019913f4998a
            └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d.jpg

86 directories, 12 files
```
After restarting the MCP Server:
```
.
├── arrange
├── completed
│   └── transfers
├── currentlyProcessing
├── DIPbackups
├── failed
├── rejected
├── sharedMicroServiceTasksConfigs
│   ├── createXmlEventsAssist
│   ├── generateAIP
│   │   └── bagit
│   ├── processingMCPConfigs
│   │   ├── automatedProcessingMCP.xml
│   │   └── defaultProcessingMCP.xml
│   └── transcoder
│       └── defaultIcons
├── SIPbackups
├── tmp
├── watchedDirectories
│   ├── activeTransfers
│   │   ├── baggitDirectory
│   │   ├── baggitZippedDirectory
│   │   ├── Dspace
│   │   ├── maildir
│   │   ├── standardTransfer
│   │   └── TRIM
│   ├── approveNormalization
│   ├── approveSubmissionDocumentationIngest
│   ├── quarantined
│   ├── SIPCreation
│   │   ├── completedTransfers
│   │   └── SIPsUnderConstruction
│   ├── storeAIP
│   ├── system
│   │   ├── autoProcessSIP
│   │   ├── autoRestructureForCompliance
│   │   ├── createAIC
│   │   └── reingestAIP
│   ├── uploadDIP
│   ├── uploadedDIPs
│   └── workFlowDecisions
│       ├── compressionAIPDecisions
│       ├── createDip
│       ├── createTree
│       ├── examineContentsChoice
│       ├── extractPackagesChoice
│       ├── metadataReminder
│       ├── quarantineTransfer
│       ├── selectFormatIDToolIngest
│       └── selectFormatIDToolTransfer
└── www
    ├── AIPsStore
    │   ├── e022
    │   │   └── 606a
    │   │       └── b9ce
    │   │           └── 480a
    │   │               └── ba3a
    │   │                   └── 0199
    │   │                       └── 13f4
    │   │                           └── 998a
    │   │                               └── 2-e022606a-b9ce-480a-ba3a-019913f4998a.7z
    │   └── transferBacklog
    │       ├── arrange
    │       └── originals
    ├── DIPsStore
    │   ├── 466c
    │   │   └── 7dc8
    │   │       └── e92d
    │   │           └── 46f2
    │   │               └── a7b7
    │   │                   └── b9af
    │   │                       └── 2e08
    │   │                           └── d990
    │   │                               └── 2-e022606a-b9ce-480a-ba3a-019913f4998a
    │   │                                   ├── METS.e022606a-b9ce-480a-ba3a-019913f4998a.xml
    │   │                                   ├── objects
    │   │                                   │   └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d-one_file.txt
    │   │                                   ├── processingMCP.xml
    │   │                                   └── thumbnails
    │   │                                       └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d.jpg
    │   └── 9b18
    │       └── b071
    │           └── b67b
    │               └── 40c9
    │                   └── 895d
    │                       └── 2f59
    │                           └── 5c26
    │                               └── 7a37
    │                                   └── 2-e022606a-b9ce-480a-ba3a-019913f4998a
    │                                       ├── METS.e022606a-b9ce-480a-ba3a-019913f4998a.xml
    │                                       ├── objects
    │                                       │   └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d-one_file.txt
    │                                       ├── processingMCP.xml
    │                                       └── thumbnails
    │                                           └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d.jpg
    └── thumbnails
        └── e022606a-b9ce-480a-ba3a-019913f4998a
            └── dc42bd70-8bc0-4ecf-b182-970fe0dae25d.jpg

86 directories, 12 files
```
So that seems to have stabilized the issue. The `Tasks` table remains empty.

**Comment by ross-spencer** [comment]
Some of the patterns existed in 1.6.1:

![image](https://user-images.githubusercontent.com/1880412/44498766-1c72b080-a646-11e8-87f8-c44e052b97b9.png)

But it seems maybe the files weren't duplicated by the various microservices running again.

* File count before: _43 directories, 9 files_
* File count after restart: _43 directories, 9 files_

**Comment by mamedin** [comment]
This bug also happens on versions:
```
archivematica_src_am_version: "v1.7.1"
archivematica_src_ss_version: "v0.11.1"
```

**Comment by ross-spencer** [comment]
In 1.6.1 the module seems to have been the AM move module:

![image](https://user-images.githubusercontent.com/1880412/44534112-a2c9d980-a6bc-11e8-9521-4b0fe1b0eba3.png)

But in 1.7.2 this has switched to copy (recursive):

![image](https://user-images.githubusercontent.com/1880412/44534173-be34e480-a6bc-11e8-8c1d-3a0bbd66e057.png)

I will try and confirm what I see in the DB, and when this happened.

**Comment by sevein** [comment]
@ross-spencer https://github.com/artefactual/archivematica/blob/353fb054bfed3cce09cc10e08068712aba2219fb/src/dashboard/src/main/migrations/0038_fix_workflow_store_dip.py#L55-L57

**Comment by sromkey** [comment]
I was able to reproduce this by choosing "Do Not Upload" followed by storing the DIP. I could not reproduce the issue by choosing "Do Not Upload" followed by "Do not store".

**Comment by ross-spencer** [comment]
Those are useful tests thanks @sromkey. I can see the introduced Microservice, `Job Handle unstored DIP` running in the one that works: 

![image](https://user-images.githubusercontent.com/1880412/44539714-859c0780-a6ca-11e8-9a4d-203d6ee33f3a.png)

**Comment by ross-spencer** [comment]
In the workflow where we can recreate this, the following Microservice tasks happen:
```
Store DIP? User choice: Store: -> 
   Retrieve DIP Storage Locations (getAipStorageLocations_v0.0 DS) -> 
      Store AIP Location (User choice from MS '%AIPsStore%') -> 
         Store DIP (storeAIP_v0.0)
```

**Comment by ross-spencer** [comment]
Summary of the issue: 

Following migration 0038 introduced in 1.7.0 there was no pathway for Archivematica to follow to remove duplicate DIP data from the uploadDIP watched directory if the content continues on to be stored by `storeAIP.py`. Being a watched directory, when the MCP Server is restarted, the chain associated with the watched dir is started again, and we see the issue logged here in #112. 

To fix this in `1.7.2` we have introduced a function in `storeAIP.py` to perform this cleardown for us. 

To test this: 

There are two-scenarios that I can identify:

1. No DIP upload but Store DIP
2. Failed DIP upload but Store DIP*

***NB.** This fix will only remove the duplicate files if it is requested that the DIP is stored. 

Output will look as follows:

**No upload DIP -> Store DIP**

![image](https://user-images.githubusercontent.com/1880412/44554124-5cdd3780-a6f5-11e8-8f00-eeb6d1c8d960.png)

**Upload DIP Failure -> Store DIP**

![image](https://user-images.githubusercontent.com/1880412/44554141-6bc3ea00-a6f5-11e8-8527-66d8ead05704.png)

Further, the tasks table in the database should show no additional entries relating to dip upload. 

* The uploadDIP directory should also be empty in both scenarios
* The uploadedDIPs directory should contain a copy of the failed upload in the failed scenario
* The storage service should have a copy of both DIPS

This PR only addresses the `stable/1.7.x` branch. A different solution might be more appropriate in `qa/1.x`.

**Comment by kellyannewithane** [comment]
Tested on a fresh install with @mamedin and it worked just fine!
